zerver/lib/upload.py

```diff
@@ -1151,11 +1150,11 @@ def create_attachment(
 
 
 def upload_message_image_from_request(
-    request: HttpRequest, user_file: File, user_profile: UserProfile
+    request: HttpRequest, user_file: File, user_profile: UserProfile, user_file_size: int
 ) -> str:
-    uploaded_file_name, uploaded_file_size, content_type = get_file_info(request, user_file)
+    uploaded_file_name, content_type = get_file_info(request, user_file)
     return upload_message_file(
-        uploaded_file_name, uploaded_file_size, content_type, user_file.read(), user_profile
+        uploaded_file_name, user_file_size, content_type, user_file.read(), user_profile
```

Comment:
Any reason not to type `user_file: UploadedFile` and use `user_file.size` here?

Reply:
Oh, because you’ve validated that it’s not `None`. Fine.

Reply:
This saves an assertion check for `user_file.size`, which is done previously in `upload_file_backend`.